be/src/storage/lake/rowset_update_state.cpp

```diff
@@ -426,29 +426,44 @@ Status RowsetUpdateState::_prepare_auto_increment_partial_update_states(uint32_t
 Status RowsetUpdateState::_prepare_partial_update_states(uint32_t segment_id, const RowsetUpdateStateParams& params,
                                                          bool need_lock) {
     CHECK_MEM_LIMIT("RowsetUpdateState::_prepare_partial_update_states");
-    std::vector<ColumnId> read_column_ids = get_read_columns_ids(params.op_write, params.tablet_schema);
 
+    // For COLUMN_UPSERT_MODE, we only need src_rss_rowids to identify new rows (UINT64_MAX)
     const auto& txn_meta = params.op_write.txn_meta();
+    bool is_column_upsert_mode = (txn_meta.partial_update_mode() == PartialUpdateMode::COLUMN_UPSERT_MODE);
+
+    std::vector<ColumnId> read_column_ids;
+    if (!is_column_upsert_mode) {
```

Comment:
Move `read_colum_ids` to the get values ​​section? if it is row mode, there will be an extra meaningless judgment and reading.

Reply:
Thanks for your time and advice.